alica_engine/src/engine/allocationauthority/CycleManager.cpp

```diff
@@ -36,13 +36,17 @@ CycleManager::CycleManager(ConfigChangeListener& configChangeListener, const Ali
         , _newestAllocationDifference(0)
         , _runningPlan(rp)
         , _myID(_teamManager.getLocalAgentID())
+        , _configChangeListener(configChangeListener)
 {
     auto reloadFunctionPtr = std::bind(&CycleManager::reload, this, std::placeholders::_1);
-    configChangeListener.subscribe(reloadFunctionPtr);
+    _reloadCallbackId = configChangeListener.subscribe(reloadFunctionPtr);
     reload(configChangeListener.getConfig());
 }
 
-CycleManager::~CycleManager() {}
+CycleManager::~CycleManager()
+{
+    _configChangeListener.unsubscribe(_reloadCallbackId);
```

Comment:
this should lead to a crash during engine termination because the plan base will be destroyed after the configChangeListener. The plan base stores a vector of shared_ptrs to running plans which if not empty is destroyed during destruction & this will crash. I think this vector won't be empty & will atleast have the master plan in it until destruction, although I'm not sure why the crash didn't happen in CI.

Anyway, I think to fix, you should empty the _runningPlans in PlanBase in stop() instead.

Reply:
Why do you think the configChangeListener will be destroyed first? AFAIK members are constructed in order of declaration (configChangeListener is declared before planBase in AlicaEngine.h) and destruction is in reverse order of construction, therefore planBase is destroyed before configChangeListener.

Reply:
ah ok then its fine